src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -349,6 +349,15 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 								  "foreign tables.")));
 	}
 
+#if (PG_VERSION_NUM >= 100000)
+	if (relation->rd_rel->relispartition)
+	{
+		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
+						errmsg("cannot distribute relation: %s", relationName),
+						errdetail("Distributing partition tables is unsupported.")));
+	}
```

Comment:
(However, agreed on the test, though I didn't want to add a test variant just for this one syntax thing @onderkalaci is likely to remove soon anyhow. That said, I'll just add an entirely new test file for PostgreSQL 10-specific syntax features).